### PR TITLE
adding outdoor and supply air temperature to furnace hwstatus parsing

### LINKED ISSUE #327: Furnace: Add OutdoorAirTemperature and SupplyAirTemperature to HWSTATUS parsing
Adding parsing of the OutdoorAirTemperature and SupplyAirTemperature to HWSTATUS, positions 52/53 and 145/146.

### PR DESCRIPTION
**Pull request by bitfliq**
closes #327 

Explicitly setting decimal precision of template sensor values to zero to match EspHome-econet sensor convention.

### CODE REVIEW

File: econet_hvac_furnace.yaml [hunk 1]
```diff
@@ -91,7 +93,7 @@ sensor:
     state_class: "measurement"
     entity_category: "diagnostic"
     filters:
-      - delta: 0.9
+      - delta: 0.91
```
Comment (barndawgie) [inline econet_hvac_furnace.yaml:96]: What's the reason for this change? Was it causing some problem?

File: econet_hvac_furnace.yaml [hunk 2]
```diff
@@ -46,6 +47,7 @@ econet:
             id(high_heat_lifetime_hrs).publish_state((x[132] << 8) + x[133]);
             id(blower_lifetime_hrs).publish_state((x[135] << 8) + x[136]);
             id(powered_lifetime_days).publish_state((x[137] << 8) + x[138]);
+            id(supply_air_temperature).publish_state(((x[145] << 8) + x[146])/10.0);
```
Comment (barndawgie) [inline econet_hvac_furnace.yaml:50]: Do you need to remove this line now that you're setting this via HWSTATUS?
```yaml
sensor_datapoint: FURNSTMP
```

File: econet_hvac_furnace.yaml [hunk 3]
```diff
@@ -29,6 +29,7 @@ econet:
             id(heat_stage).publish_state(x[26]);
             id(static_pressure).publish_state(((x[15] << 8) + x[16]) / 5280.0);
             id(return_air_temperature).publish_state(((x[50] << 8) + x[51]) / 10.0);
+            id(outdoor_air_temperature).publish_state(((x[52] << 8) + x[53])/10.0);
```
Comment (barndawgie) [inline econet_hvac_furnace.yaml:32]: Does this value only exist/get data if there is an ODU? If so, does the corresponding sensor need to somehow move to the ODU YAML?

**Review by barndawgie — CHANGES_REQUESTED** [review]
A few comments in line. I'm also wondering:
* Have you tested this with your system at home?
* Your changes are only to the furnace YAML. Would they potentially also apply to Air Handlers?

File: econet_hvac_furnace.yaml (hunk 3 above)
Comment (justbruce) [reply econet_hvac_furnace.yaml:32]: All of the furnaces have an input for an Outdoor Air Temp thermistor. Unsure on air handlers.  Seems this value can get updated in multiple ways, maybe

Comment (bitfliq) [reply econet_hvac_furnace.yaml:32]: Expanding on @justbruce comment - in my case this comes from a thermistor located on the heat pump outside, but it is read by the furnace hardware.

File: econet_hvac_furnace.yaml (hunk 1 above)
Comment (bitfliq) [reply econet_hvac_furnace.yaml:96]: Digging in a little bit more, there's a default precision in the econet sensor here (0 decimals) that makes my change unnecessary.    For the template sensors the default precision is [1 decimal ](https://github.com/esphome/esphome/blob/c35a21773e3158484c60879a8fdcb806b9cc1163/esphome/components/template/sensor/__init__.py#L19), which led to some odd behaviors. I'll set it back to 0.9 and make the accuracy_decimals 0 for the templates.

```
[12:39:52][V][sensor:043]: 'Supply Air Temperature': Received new state 105.300003
[12:39:52][D][sensor:094]: 'Supply Air Temperature': Sending state 105.30000 °F with 0 decimals of accuracy
```

```
[12:39:55][V][sensor:043]: 'Supply Air Temperature': Received new state 106.099998
[12:39:55][D][sensor:094]: 'Supply Air Temperature': Sending state 106.10000 °F with 1 decimals of accuracy
```

File: econet_hvac_furnace.yaml (hunk 2 above)
Comment (bitfliq) [reply econet_hvac_furnace.yaml:50]: I think other folks have expressed a preference for the datapoint querying directly.  I'll comment that out and convert it to a 'template' instead of an 'econet' sensor.

Comment (bitfliq) [reply econet_hvac_furnace.yaml:50]: Left this datapoint in but set to request_mod:none. I believe this will let it publish updates if appropriate but not request.

**Comment by bitfliq** [comment]
>     Have you tested this with your system at home?
yes, works for my system.

>     Your changes are only to the furnace YAML. Would they potentially also apply to Air Handlers?
I don't think so. Those would be handled by changes to AIRHSTAT parsing.

File: econet_hvac_furnace.yaml (hunk 2 above)
Comment (barndawgie) [reply econet_hvac_furnace.yaml:50]: I just posted in Discord since I'm not sure I really understand this. I would be curious to hear @stockmopar take on whether this is the right approach.

**Comment by barndawgie** [comment]
These look good to me, but it might be good to get @stockmopar's eyes on it as well.

**Comment by tronikos** [comment]
Update PR title and description to match changes

**Comment by bitfliq** [comment]
@tronikos happy to change title/description to whatever you envision - can you clarify what you're looking for?

**Comment by tronikos** [comment]
Just to match the actual latest changes included in the PR. e.g. "modified delta to be 0.91 degrees to avoid hysteresis" doesn't apply anymore or "1 decimal" doesn't seem to match the actual change.

**Comment by bitfliq** [comment]
> Just to match the actual latest changes included in the PR. e.g. "modified delta to be 0.91 degrees to avoid hysteresis" doesn't apply anymore or "1 decimal" doesn't seem to match the actual change.

done, thanks for pointing this out
